File: game_plugins/hase_und_igel_new/shared/sc/plugin2018/WelcomeMessage.java
package sc.plugin2018;

import com.thoughtworks.xstream.annotations.XStreamAlias;
import sc.shared.PlayerColor;

@XStreamAlias(value="welcomeMessage")
public class WelcomeMessage
{
	private PlayerColor color;
	
	public WelcomeMessage(PlayerColor color)
	{
		this.color = color;
	}
	
	public PlayerColor getPlayerColor()
	{
		return color;
	}
}

File: sdk/src/framework/sc/shared/WelcomeMessage.kt
package sc.shared

import com.thoughtworks.xstream.annotations.XStreamAlias
import com.thoughtworks.xstream.annotations.XStreamAsAttribute
import sc.api.plugins.ITeam
import sc.protocol.room.RoomMessage

/** Nachricht, die zu Beginn eines Spiels an einen Client geschickt wird, um ihm seine Spielerfarbe mitzuteilen.  */
@Suppress("DataClassPrivateConstructor")
@XStreamAlias(value = "welcomeMessage")
data class WelcomeMessage private constructor(
        @XStreamAsAttribute val color: String
): RoomMessage {
    constructor(color: ITeam): this(color.name)
}